src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -276,8 +282,8 @@ public void appendLabel(String key, String value) {
     public ComputeEngineInstance provision(TaskListener listener) throws IOException {
         PrintStream logger = listener.getLogger();
         try {
-            Instance i = instance();
-            Operation operation = cloud.client.insertInstance(cloud.projectId, i);
+            Instance instance = instance();
+            Operation operation = cloud.client.insertInstance(cloud.projectId, Optional.ofNullable(template), instance);
```

Comment:
Could you do me a favor and please add a TODO comment here referencing this issue: JENKINS-55285